src/main/java/org/apache/commons/lang3/ThreadUtils.java

```diff
@@ -0,0 +1,341 @@
+        int threadGroupCount = 0;
+        ThreadGroup[] threadGroups;
+        do {
+            estimatedThreadGroupCount *= 2;
+            threadGroups = new ThreadGroup[estimatedThreadGroupCount];
+            threadGroupCount = systemThreadGroup.enumerate(threadGroups);
+        } while (threadGroupCount == estimatedThreadGroupCount);
+
+        final ThreadGroup[] allGroups = new ThreadGroup[threadGroupCount+1];
+        allGroups[0] = systemThreadGroup;
+        System.arraycopy(threadGroups, 0, allGroups, 1, threadGroupCount);
+        return allGroups;
+    }
+
+    /**
+     * Return the system thread group (sometimes also referred as "root thread group")
+     *
+     * @return the system thread group
+     * @throws  SecurityException  if the current thread cannot modify
+     *          thread groups from this thread's thread group up to the system thread group
+     */
+    public static ThreadGroup getSystemThreadGroup() {
+        ThreadGroup threadGroup = Thread.currentThread().getThreadGroup();
+        while(threadGroup.getParent() != null) {
+            threadGroup = threadGroup.getParent();
+        }
+        return threadGroup;
+    }
+
+    /**
+     * Return all active threads (A thread is active if it has been started and has not yet died)
+     *
+     * @return all active threads
+     * @throws  SecurityException
+     *          if the current thread cannot access the system thread group
+     *
+     * @throws  SecurityException  if the current thread cannot modify
+     *          thread groups from this thread's thread group up to the system thread group
+     */
+    public static Thread[] getAllThreads() {
+        final ThreadGroup systemThreadGroup = getSystemThreadGroup();
+        int estimatedThreadCount = systemThreadGroup.activeCount();
+        int threadCount = 0;
+        Thread[] threads;
+        do {
+            estimatedThreadCount *= 2;
+            threads = new Thread[estimatedThreadCount];
+            threadCount = systemThreadGroup.enumerate(threads);
+        } while (threadCount == estimatedThreadCount);
+
+        return Arrays.copyOf(threads, threadCount);
+    }
+
+    /**
+     * Return actve threads with the specified name
+     *
+     * @param threadName The thread name
+     * @return The threads with the specified name or an empty array if no such thread exists
+     * @throws IllegalArgumentException if the specified name is null
+     * @throws  SecurityException
+     *          if the current thread cannot access the system thread group
+     *
+     * @throws  SecurityException  if the current thread cannot modify
+     *          thread groups from this thread's thread group up to the system thread group
+     */
+    public static Thread[] findThreadsByName(final String threadName) {
+        if (threadName == null) {
+            throw new IllegalArgumentException("The threadName must not be null");
+        }
+        final Thread[] allThreads = getAllThreads();
```

Comment:
this allocation seems wasteful, couldnt we just inline the allthreads call and directly check them?